### ISSUE TITLE
Intelegent teleport

### ISSUE THREAD

**Issue by mennas** [PocketMobsters/PokeMobBot#252]
We knew that if we tele more than 100m well get softban, why not add feature like this ; 
-if next pokestop is <=100m bot will use tele (if tele set to true)
-if next poke stop is >=101m bot will walk (the speed is based on "walking speed value")

I usually set my tele to true for quicker leveling

**Comment by Fenixxia** [comment]
Действительно, интеллигентно. Мне нравится.

**Comment by lancerts** [comment]
I have also noticed that. The other thing that can be done is try to catch pokemon without constantly visiting pokestops. Only if pokeballs decrease certain value, then farm pokestops. Otherwise, we spend too much time in unban (with teleport=TRUE).

**Comment by nothingtoseeheremates** [comment]
Yeah i suggested that already, i think better than that will be set user locations to go from, so you set on config original coords, and then if it doesn't find pokestops in a distance greater than 200 meters teleport to somewhere else fresh with pokestops not on cooldown, so like you would go from the uk to central park in the us etc, and then return to original when that condition was met again, you're gonna get a full softban on a teleport around 200 meters away anyway so may as well go somewhere fresh with a bunch of pokestops off cooldown.

Either that or make the bot teleport in smaller increments if distance greater than 100 to avoid softban.

Cheers

**Comment by OverCraftX** [comment]
I think its easy to find how many time do you need to teleport to a x meter away. Its very constant. So if the pokestop its 200meters away, first try to find another one closer, if not, just wait a few seconds before the jump (maybe usefull for sniping) and then jump.

Anyways its similar to wait for the pokestop to be "unban", probably that the distance/timeout.

**Comment by nothingtoseeheremates** [comment]
Yeah i think you need to just teleport away, just tried a version that did small teleport increments and you will still get hit with softbans due to the bot just jumping to pokestops 800m away after a while, and then will jump again shortly 900m etc, so i think the future and biggest xp increase would be setting more than 1 coords where the bot can cycle through to farm, exact same softban time as jumping to a pokestop 800m away, does it once only per each farming spot change, and since it will arrive in a fresh place every time the xp/h would be greatly improved.
